plugins/outputs/pusher/pusher.go

```diff
@@ -0,0 +1,95 @@
+package pusher
+
+import (
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers"
+	"github.com/pusher/pusher-http-go"
+)
+
+type Pusher struct {
+	AppId       string `toml:"app_id"`
+	AppKey      string `toml:"app_key"`
+	AppSecret   string `toml:"app_secret"`
+	ChannelName string `toml:"channel_name"`
+
+	Host string `toml:"host"`
+
+	Secure bool `toml:"secure"`
+
+	client *pusher.Client
+
+	serializer serializers.Serializer
+}
+
+var sampleConfig = `
+  ## Pusher Credentials
```

Comment:
can you explain more about how the credentials work? Which of the options below are required?